File: session-04-proxy-and-aop/typescript/src/order/order.service.ts
import {Injectable} from "@nestjs/common";
import {Order} from "./order.model";
import {OrderRepository} from "./order.repository";

// Notice: NO logging, NO timing code here. Interceptors add it.
@Injectable()
export class OrderService {
    constructor(private readonly repository: OrderRepository) {
    }

    listOrders(): Order[] {
        return this.repository.findAll();
    }

    findOrder(id: number): Order | undefined {
        return this.repository.findById(id);
    }
}

File: session-04-proxy-and-aop/java/src/main/java/coffeeshop/order/OrderService.java
package coffeeshop.order;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

// Notice: NO logging, NO timing, NO audit code here.
// All cross-cutting concerns are added by proxies/aspects.
@Service
public class OrderService {

    private final OrderRepository repository;

    public OrderService(OrderRepository repository) {
        this.repository = repository;
    }

    public List<Order> listOrders() {
        return repository.findAll();
    }

    public Optional<Order> findOrder(int id) {
        return repository.findById(id);
    }
}
